demos/list.html

```diff
@@ -1330,47 +1330,34 @@ <h3>Example - Interactive List (Dark Theme)</h3>
           });
         }
 
-        var deactivateRippleHandler = null;
-        var checkboxListRippleSurfaceClass = 'checkbox-list-ripple-surface';
-
         function clickHandler(evt, rippleFoundation, li) {
           var checkbox = li.querySelector('.mdc-checkbox__native-control');
-          var isLabel = Array.prototype.some.call(checkbox.labels, function(label) { return label === evt.target; });
-          if(evt.target !== checkbox && !isLabel) {
+          var label = li.querySelector('label');
+          if(evt.target !== checkbox && evt.target !== label) {
             checkbox.checked = !checkbox.checked;
-            li.classList.add(checkboxListRippleSurfaceClass);
-            rippleFoundation.activate();
-
-            deactivateRippleHandler = e => deactivateRipple(e, rippleFoundation, li);
-            li.addEventListener('mouseup', deactivateRippleHandler);
           }
-        };
-
-        function deactivateRipple(evt, rippleFoundation, li) {
-          li.classList.remove(checkboxListRippleSurfaceClass);
-          rippleFoundation.deactivate();
-          li.removeEventListener('mouseup', deactivateRippleHandler);
+          if(evt.target !== checkbox) {
+            rippleFoundation.activate(evt);
+          }
         };
 
         function addRippleToListItems(list) {
-          list.querySelectorAll('.mdc-list-item').forEach(function(li) {
-            var rippleAdapter = mdc.ripple.MDCRipple.createAdapter({
-              unbounded: false,
-              disabled: false,
-              root_: li
-            });
-
-            var rippleFoundation = new mdc.ripple.MDCRippleFoundation(rippleAdapter);
-            rippleFoundation.layout();
+          var listItems = list.querySelectorAll('.mdc-list-item');
+          [].forEach.call(listItems, function(li) {
+            var rippleFoundation = new mdc.ripple.MDCRippleFoundation(
+              mdc.ripple.MDCRipple.createAdapter({
+                unbounded: false, disabled: false, root_: li
+              })
```

Comment:
Add a `;`

Reply:
This is inside the new mdc.ripple.MDCRippleFoundation function.

Reply:
Oh woops! You're absolutely right 😀